### PR TITLE
Tune Hackney to not overload WebDrivers

### PR DESCRIPTION
**Pull request by mhanberg**
Hopefully this resolves/helps #365 

I have tested this locally with hundreds to thousands of tests running in parallel, and the only reason I've seen it fall apart is because of my computer running out of open file handles.

### CODE REVIEW

**Comment by michallepicki** [comment]
> I have tested this locally with hundreds to thousands of tests running in parallel, and the only reason I've seen it fall apart is because of my computer running out of open file handles.

Sounds good, but are you able to reproduce the error from #365 when running without that change?

File: lib/wallaby.ex [hunk 1]
```diff
@@ -32,6 +32,7 @@ defmodule Wallaby do
     children = [
       supervisor(Wallaby.Driver.ProcessWorkspace.ServerSupervisor, []),
       supervisor(driver(), [[name: Wallaby.Driver.Supervisor]]),
+      :hackney_pool.child_spec(:wallaby_pool, [timeout: 15_000, max_connections: 4]),
```
Comment (michallepicki) [inline lib/wallaby.ex]: I could not find docs on that in hackney, what is this timeout for? Is it a timeout for starting a worker or does it set default http request timeout for a worker?

Comment (keathley) [reply lib/wallaby.ex]: The timeout here is how long the connection will be kept alive in the pool.

Comment (michallepicki) [reply lib/wallaby.ex]: @keathley Doesn't it prevent the other configuration from being applied then? Sorry if asking stupid questions, never had to tweak hackney
```
config :wallaby,
  hackney_options: [timeout: :infinity, recv_timeout: :infinity]
```

Comment (keathley) [reply lib/wallaby.ex]: No worries :). Both of those options are for each individual request. The timeout option that is being added in this PR sets how long the TCP connection should be kept alive in-between requests.

**Comment by mhanberg** [comment]
> > I have tested this locally with hundreds to thousands of tests running in parallel, and the only reason I've seen it fall apart is because of my computer running out of open file handles.
> 
> Sounds good, but are you able to reproduce the error from #365 when running without that change?

Yes 👍

**Comment by michallepicki** [comment]
> > > I have tested this locally with hundreds to thousands of tests running in parallel, and the only reason I've seen it fall apart is because of my computer running out of open file handles.
> > 
> > 
> > Sounds good, but are you able to reproduce the error from #365 when running without that change?
> 
> Yes +1

Oh god I may be able to finally speed up my 1 hour long test runs

**Pull request merged**